src/arviz_plots/plot_matrix.py

```diff
@@ -174,10 +218,29 @@ def get_target(self, var_name, selection, var_name_y=None, selection_y=None):
             Mapping with with coordinate subset along the y dimension.
             If not provided it will be assumed as being `selection`
         """
+        if self.orientation == "diagonal":
+            var_name_x = var_name
+            var_name_y = var_name
+            selection_x = selection
+            selection_y = selection
+        elif self.orientation == "row":
+            var_name_x = var_name
+            var_name_y = self.fixed_var_name
+            selection_x = selection
+            selection_y = self.fixed_selection
+        elif self.orientation == "col":
+            var_name_x = self.fixed_var_name
+            var_name_y = var_name
+            selection_x = self.fixed_selection
+            selection_y = selection
+        else:
+            var_name_x = var_name
+            selection_x = selection
```

Comment:
Given the behaviour of `subset_matrix_da` I think this means that the simple fact of `var_name_y` and `selection_y` being None will already behave like "diagonal" even if the `orientation` attribute is not None. I think it is actually fine and we only need to check orientation `row` or `col`, with no orientation meaning "diagonal" which I think is a good default.